de.metas.contracts/src/main/java/de/metas/contracts/subscription/impl/SubscriptionService.java

```diff
@@ -84,14 +84,17 @@ public static int changeRecipient(@NonNull final ChangeRecipientsRequest changeR
 		int DropShip_Location_ID;
 		int DropShip_User_ID;
 
+		boolean IsPermanentRecipient;
+
 		@Builder
 		private ChangeRecipientsRequest(
 				@NonNull final I_C_Flatrate_Term term,
 				@NonNull final Timestamp dateFrom,
 				@NonNull final Timestamp dateTo,
 				final int DropShip_BPartner_ID,
 				final int DropShip_Location_ID,
-				final int DropShip_User_ID)
+				final int DropShip_User_ID,
+				boolean IsPermanentRecipient)
```

Comment:
afaiu, here you could assert that IsPermanentRecipient needs to be true, or dateTo needs to be != null